crypto/ess/ess_lib.c

```diff
@@ -273,3 +274,86 @@ int ESS_SIGNING_CERT_V2_add(PKCS7_SIGNER_INFO *si,
     OPENSSL_free(pp);
     return 0;
 }
+
+static int ess_issuer_serial_cmp(const ESS_ISSUER_SERIAL *is, const X509 *cert)
+{
+    GENERAL_NAME *issuer;
+
+    if (is == NULL || cert == NULL || sk_GENERAL_NAME_num(is->issuer) != 1)
+        return -1;
+
+    issuer = sk_GENERAL_NAME_value(is->issuer, 0);
+    if (issuer->type != GEN_DIRNAME
+        || X509_NAME_cmp(issuer->d.dirn, X509_get_issuer_name(cert)) != 0)
+        return -1;
+
+    if (ASN1_INTEGER_cmp(is->serial, X509_get_serialNumber((X509 *)cert)) != 0)
```

Comment:
Would it make sense to just `return ASN1_INTEGER_cmp(is->serial, X509_get_serialNumber((X509 *)cert));` here?